snakemake/logging.py

```
@@ -442,293 +908,112 @@ def d3dag(self, **msg):
+                show_failed_logs=show_failed_logs,
+                printshellcmds=printshellcmds,
+            )
+        )
+        logfile_handler.addFilter(DefaultFilter(quiet=quiet, debug_dag=debug_dag))
+        logfile_handler.setLevel(logging.DEBUG if verbose else logging.INFO)
+        logfile_handler.name = "DefaultLogFileHandler"
+        logger.addHandler(logfile_handler)
 
 
 def setup_logger(
     handler=[],
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)